src/view/treeNodes/treeNode.ts

```diff
@@ -46,7 +48,18 @@ export abstract class TreeNode implements vscode.Disposable {
 		return this.parent.refresh(treeNode);
 	}
 
+	async cachedChildren(): Promise<TreeNode[]> {
+		if (this.children && this.children.length) {
+			return this.children;
+		}
+		return this.getChildren();
+	}
+
 	async getChildren(): Promise<TreeNode[]> {
+		if (this.children && this.children.length) {
+			dispose(this.children);
```

Comment:
Hey @alexr00, sorry for digging up an old PR, but I've got a quick question.

Whilst debugging something for an upcoming PR I ran into this. Why do we dispose of the children here within `getChildren`?

Reply:
With the introduction of `cachedChildren`, we started caching the child nodes. So, when calling `getChildren` instead of `cachedChildren`, we want to recalculate all the child nodes. We should also clear the cache at this time so that we don't end up with old child nodes sticking around. Since the child nodes are disposable, they should also be disposed when they are cleared.